services/functions/linkdex-cid.ts

```diff
@@ -16,13 +17,19 @@ export interface Report extends LDReport {
   cars: string[]
 }
 
+export interface ReporterOptions {
+  /** Hash blocks while indexing to verify block bytes match the CID. */
+  hash?: boolean
+}
+
 const nullReport: Report = { cars: [], structure: 'Unknown', blocksIndexed: 0, uniqueCids: 0, undecodeable: 0 }
 
 export const handler: APIGatewayProxyHandlerV2 = async (event) => {
   const cid = event.rawPath.split('/cid/')[1]
+  const hash = event.rawQueryString.includes('hash=true')
   const bucket = process.env.BUCKET_NAME ?? ''
   const s3 = new S3Client({})
-  const reporters = [new CompleteReporter(bucket, s3), new RawReporter(bucket, s3)]
+  const reporters = [new CompleteReporter(bucket, s3, { hash }), new RawReporter(bucket, s3, { hash })]
```

Comment:
minor but maybe a little nicer to do the query wrangling once and pass in the chosen indexer to a reporter.
```suggestion
  const index = this.options.hash ? new HashingLinkIndexer() : new LinkIndexer()
  const reporters = [new CompleteReporter(bucket, s3, index), new RawReporter(bucket, s3, index)]
```

Reply:
The `RawReporter` needs to create an indexer for each group of CARs it considers so we cannot pass here.